src/Symfony/Component/Config/Definition/Dumper/XmlReferenceDumper.php

```diff
@@ -227,10 +227,6 @@ private function writeNode(NodeInterface $node, int $depth = 0, bool $root = fal
 
                 if ($attributesCount === $i++) {
                     $this->writeLine($rootIsEmptyTag ? '/>' : '>', $depth);
-
-                    if ($rootIsVariablePrototype) {
```

Comment:
This seems to indicate a bigger bug to me, because this looks like a case where we could still end up with no end tag

Reply:
I will confirm that with a test and fix it, should be done in another PR ?

Reply:
on another PR yes if there is an issue, because it's likely going to be for 5.4

Reply:
I reverted this part